client/src/controllers/SwapController.ts

```diff
@@ -0,0 +1,288 @@
+      if (form.hasAttribute('data-controller')) return;
+
+      // set up the controller on the form and the action on the inputs
+
+      form.setAttribute('data-controller', identifier);
+      form.setAttribute(
+        'data-action',
+        [
+          `change->${identifier}#updateLocation`,
+          `input->${identifier}#updateLocation`,
+        ].join(' '),
+      );
+      input.setAttribute(`data-${identifier}-target`, 'input');
+    });
+  }
+
+  connect() {
+    // set up values that may be set with the global header approach
+    const formContainer = this.hasInputTarget
+      ? this.inputTarget.form
+      : this.element;
+    const { targetOutput, url: src } = getGlobalHeaderSearchOptions();
+    this.srcValue =
+      src || this.srcValue || formContainer?.getAttribute('action') || '';
+    this.targetValue = targetOutput || this.targetValue;
+    this.targetElement = this.getTarget(this.targetValue);
+
+    // set up icons
+    this.iconElement = null;
+    const iconContainer = (
+      this.hasInputTarget ? this.inputTarget : this.element
+    ).parentElement;
+
+    this.iconElement = iconContainer?.querySelector('use') || null;
+    this.iconValue = this.iconElement?.getAttribute('href') || '';
+
+    // set up initial loading state (if set originally in the HTML)
+    this.loadingValue = false;
+
+    // set up debounced update method
+    this.updateLocation = debounce(
+      this.swapLocation.bind(this),
+      this.waitValue,
+    );
+  }
+
+  getTarget(targetValue = this.targetValue) {
+    const targetElement = document.querySelector(targetValue);
+
+    const foundTarget = targetElement && targetElement instanceof HTMLElement;
+    const hasValidUrlValue = !!this.srcValue;
+
+    const errors: string[] = [];
+
+    if (!foundTarget) {
+      errors.push(`Cannot find valid target element at "${targetValue}"`);
+    }
+
+    if (!hasValidUrlValue) {
+      errors.push(`Cannot find valid src URL value`);
+    }
+
+    if (errors.length) {
+      throw new Error(errors.join(', '));
+    }
+
+    return targetElement as HTMLElement;
+  }
+
+  /**
+   * Toggle the visual spinner icon if available and ensure content about
+   * to be replaced is flagged as busy.
+   */
+  loadingValueChanged(isLoading: boolean) {
+    if (isLoading) {
+      this.targetElement?.setAttribute('aria-busy', 'true');
+      this.iconElement?.setAttribute('href', '#icon-spinner');
+    } else {
+      this.targetElement?.removeAttribute('aria-busy');
+      this.iconElement?.setAttribute('href', this.iconValue);
+    }
+  }
+
+  /**
+   * Abort any existing requests & set up new abort controller, then fetch and patch
+   * in the HTML results, dispatching an event and handling any clean up.
+   * Cancel any in progress results request using the AbortController so that
+   * a faster response does not replace an in flight request.
+   */
+  async request(param: string | CustomEvent<{ url: string }>) {
```

Comment:
This seems like a useful method that can be used as the Stimulus action when you just need to patch the DOM without updating the location. Perhaps rename this to `patch()` (or `swap()`, but that might be confusing), as this method does more than just firing a request?